### PR TITLE
add v018 migrate

### PR DESCRIPTION
**Pull request by zhshy11**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/genutil/client/cli/migrate.go [hunk 1]
```diff
@@ -12,9 +12,11 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/version"
 	extypes "github.com/cosmos/cosmos-sdk/x/genutil"
+	v017 "github.com/okex/exchain/x/genutil/client/legacy/v0_17"
 )
 
 var migrationMap = extypes.MigrationMap{
+	"v017": v017.Migrate,
```
Comment (banishee) [inline x/genutil/client/cli/migrate.go]: v0.17

File: x/genutil/client/legacy/v0_17/migrate.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+// Migrate migrates exported state from v0.16 to a v0.17 genesis state.
+func Migrate(appState genutil.AppMap) genutil.AppMap {
+	v016Codec := codec.New()
+	codec.RegisterCrypto(v016Codec)
+
+	v017Codec := codec.New()
+	codec.RegisterCrypto(v017Codec)
+
+	// migrate auth state
```
Comment (banishee) [inline x/genutil/client/legacy/v0_17/migrate.go]: change comment

File: x/genutil/client/legacy/v0_17/migrate.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	if appState[v017evm.ModuleName] != nil {
+		var evmState v016evm.GenesisState
+		v016Codec.MustUnmarshalJSON(appState[v017evm.ModuleName], &evmState)
+
+		delete(appState, v017evm.ModuleName) // delete old key in case the name changed
+		appState[v017evm.ModuleName] = v017Codec.MustMarshalJSON(v017evm.Migrate(evmState))
+	}
+
+	// migrate auth state
```
Comment (banishee) [inline x/genutil/client/legacy/v0_17/migrate.go]: change comment

File: x/evm/legacy/v0_17/Migrate.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+const (
+	ModuleName = "evm"
+)
+
+// Migrate adds contract
+func Migrate(oldGenState v0_16.GenesisState) GenesisState {
+	params := Params{
+		EnableCreate:                      oldGenState.Params.EnableCreate,
```
Comment (banishee) [inline x/evm/legacy/v0_17/Migrate.go]: it should be false

**Review by banishee — APPROVED** [review]

File: x/genutil/client/legacy/v0_17/migrate.go (hunk 2 above)
Comment (zhshy11) [reply x/genutil/client/legacy/v0_17/migrate.go]: done

File: x/genutil/client/cli/migrate.go (hunk 1 above)
Comment (zhshy11) [reply x/genutil/client/cli/migrate.go]: done，to v0.18

File: x/genutil/client/legacy/v0_17/migrate.go (hunk 3 above)
Comment (zhshy11) [reply x/genutil/client/legacy/v0_17/migrate.go]: done

File: x/evm/legacy/v0_17/Migrate.go (hunk 4 above)
Comment (zhshy11) [reply x/evm/legacy/v0_17/Migrate.go]: done

**Pull request merged**